Analysis/include/Coda3EventDecoder.h

```diff
@@ -0,0 +1,112 @@
+#ifndef	CODA3EVENTDECODER_H
+#define CODA3EVENTDECODER_H
+
+#include "VEventDecoder.h"
+#include "Rtypes.h"
+
+#include <vector>
+
+class Coda3EventDecoder : public VEventDecoder
+{
+public:
+		Coda3EventDecoder() : fControlEventFlag(kFALSE), TSROCNumber(0) { }
+		~Coda3EventDecoder() { }
+public:
+// Encoding Functions
+	virtual std::vector<UInt_t> EncodePHYSEventHeader();
+	virtual void EncodePrestartEventHeader(int* buffer, int buffer_size, int runnumber, int runtype = 0);
+  virtual void EncodeGoEventHeader(int* buffer, int buffer_size);
+  virtual void EncodePauseEventHeader(int* buffer, int buffer_size);
+  virtual void EncodeEndEventHeader(int* buffer, int buffer_size);
+
+public:
+// Decoding Functions
+  virtual Int_t DecodeEventIDBank(UInt_t *buffer);
+public:
+// Boolean Functions
+	virtual Bool_t IsPhysicsEvent();
+private:
+// Debugging Functions
+	void printUserEvent(const UInt_t *buffer);
+	virtual void PrintDecoderInfo(QwLog& out);
+protected:
+	UInt_t InterpretBankTag(UInt_t tag);
+	Int_t trigBankDecode(UInt_t* buffer);
+	void trigBankErrorHandler( Int_t flag );
```

Comment:
trigBankDecode has a try-catch runtime exception setup. Question is, what do we do when we have a bad trigger bank? We can't return a non-"CODA_OK" value because that causes QwParity to stop acquiring new events (eventbuffer.GetNextEvent only checks if it returns CODA_OK to determine if it should continue getting events). 

This function is a prototype to remind us to figure out how to handle events with bad trig banks. 
Coda 2 lets bad events pass through and lets the subsystems handle them (if they aren't physics nor EPICS nor ROCConfig, then the event is essentially skipped w/o being processed). Following that model, the function is currently implemented to set all flags to false and all decoder variables to 0. 

Untested (how does one make bad data? -- QwMockDataGenerator works too well).

Reply:
Simply overriding the "Encode" functions to generate bad fake data would work